portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns a random greeting. */
+@WebServlet(urlPatterns = "/random-greeting")
+public final class RandomGreetingServlet extends HttpServlet {
+
+  private List<String> greetings;
```

Comment:
If you have Google Guava already as a project dependency, then ImmutableList is a slightly better fit than List or ArrayList here. You'll know if you do or not by including the `import` and then recompiling to make sure that works. If you don't have it yet, then skip this comment for now... I think adding dependencies to maven is a separate unit.

```java
import com.google.common.collect.ImmutableList;
```

```java
  private ImmutableList<String> greetings;
```

```java
  @Override
  public void init() {
    greetings = ImmutableList.of(
        "Hello!",
        "Greetings!",
        "Welcome!",
        "How do you do?",
        "Hi!");
  }
```

Reply:
Does not compile :(
package com.google.common.collect does not exist

Reply:
Acknowledged, resolving